packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -339,25 +339,30 @@ class AppDomain extends Domain {
       );
     }
 
+    bool supportsRestart = hotMode ? device.supportsHotMode : device.supportsRestart;
```

Comment:
When a user issues a reload, that will be when the state could change from supported ==> not supported? The response from an app.restart command is currently just a boolean. We could change that to a more structured response: bool for success, a message if there's a failure (`Unsupported change to ...`), and another bool to say whether any future reloads will succeed. It would then be up to the user (or the IDE) to issue a restart in order to reset the hot reload state.